src/kademlia/dht_storage.cpp

```diff
@@ -178,9 +178,21 @@ namespace
 		// explicitly disallow assignment, to silence msvc warning
 		immutable_item_comparator& operator=(immutable_item_comparator const&);
 
-		node_id const& m_our_id;
+		std::vector<node_id> const& m_node_ids;
```

Comment:
this is a reference, and it's just pointing to the existing member of the storage. I don't see any heap allocation here.